cng/tls1prf.go

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//go:build windows
+// +build windows
+
+package cng
+
+import (
+	"errors"
+	"hash"
+	"unsafe"
+
+	"github.com/microsoft/go-crypto-winnative/internal/bcrypt"
+)
+
+func loadTLS1PRF(id string) (bcrypt.ALG_HANDLE, error) {
+	h, err := loadOrStoreAlg(id, 0, "", func(h bcrypt.ALG_HANDLE) (interface{}, error) {
+		return h, nil
+	})
+	if err != nil {
+		return 0, err
+	}
+	return h.(bcrypt.ALG_HANDLE), nil
+}
+
+func TLS1PRF(secret, label, seed []byte, keyLen int, h func() hash.Hash) ([]byte, error) {
+	algID := bcrypt.TLS1_1_KDF_ALGORITHM
+	var hashID string
+	if h != nil {
+		// TLS 1.0/1.1 PRF doesn't allow to specify the hash function,
```

Comment:
Nit: might want to move this comment above, it was kind of confusing as its explaining what just happened vs what's about to happen (also possible I'm just being slow today, but it took me a minute to understand)

Reply:
Hmm. I would probably go further and negate it and split it up so it describes what's happening:

```go
func TLS1PRF(secret, label, seed []byte, keyLen int, h func() hash.Hash) ([]byte, error) {
	// TLS 1.0/1.1 PRF uses MD5SHA1.
	algID := bcrypt.TLS1_1_KDF_ALGORITHM
	var hashID string
	if h != nil {
		// If h is specified, assume the caller wants to use TLS 1.2 PRF.
		// TLS 1.0/1.1 PRF doesn't allow specifying the hash function.
		if hashID = hashToID(h()); hashID == "" {
			return nil, errors.New("cng: unsupported hash function")
		}
		algID = bcrypt.TLS1_2_KDF_ALGORITHM
	}
```